app/controllers/concerns/create_avis_concern.rb

```diff
@@ -10,28 +10,42 @@ def create_avis_from_params(dossier, confidentiel = false)
     # the :emails parameter is a 1-element array.
     # Hence the call to first
     # https://github.com/rails/rails/issues/17225
-    emails = create_avis_params[:emails].first.split(',').map(&:strip)
+    expert_emails = create_avis_params[:emails].first.split(',').map(&:strip)
+    allowed_dossiers = [dossier]
+
+    if create_avis_params[:invite_linked_dossiers].present?
+      allowed_dossiers += dossier.linked_dossiers
+    end
 
     create_results = Avis.create(
-      emails.map do |email|
-        {
-          email: email,
-          introduction: create_avis_params[:introduction],
-          claimant: current_instructeur,
-          dossier: dossier,
-          confidentiel: confidentiel
-        }
+      expert_emails.flat_map do |email|
+        allowed_dossiers.map do |dossier|
+          {
+            email: email,
+            introduction: create_avis_params[:introduction],
+            claimant: current_instructeur,
+            dossier: dossier,
+            confidentiel: confidentiel
+          }
+        end
       end
     )
 
     persisted, failed = create_results.partition(&:persisted?)
 
     if persisted.any?
-      sent_emails_addresses = persisted.map(&:email_to_display).join(", ")
-      flash.notice = "Une demande d'avis a été envoyée à #{sent_emails_addresses}"
+      sent_emails_addresses = []
       persisted.each do |avis|
-        dossier.demander_un_avis!(avis)
+        linked_dossier = avis.dossier != dossier
```

Comment:
```suggestion
        is_linked_dossier = avis.dossier != dossier
```

(ce n'est pas le dossier lié, c'est "est-ce qu'il est lié ?")